src/test/java/com/iexec/worker/tee/post/compute/web2/DropBoxServiceTests.java

```diff
@@ -0,0 +1,83 @@
+package com.iexec.worker.tee.post.compute.web2;
+
+import com.dropbox.core.DbxException;
+import com.dropbox.core.v2.DbxClientV2;
+import com.dropbox.core.v2.files.*;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class DropBoxServiceTests {
+    private final static String DROPBOX_PATH = "DROPBOX_PATH";
+
+    @TempDir
+    File tmpFolder;
+
+    DropBoxService dropBoxService = new DropBoxService();
+
+    @Test
+    void shouldUploadFile() throws IOException, DbxException {
+        final File fileToUpload = Files.writeString(Path.of(tmpFolder.getAbsolutePath(), "fileToUpload.zip"), "Some dummy content").toFile();
+        final String pathDisplay = "https://foo.bar/fileToUpload.zip";
+
+        final DbxClientV2 client = mock(DbxClientV2.class);
+        final DbxUserFilesRequests files = mock(DbxUserFilesRequests.class);
+        final UploadBuilder uploadBuilder = mock(UploadBuilder.class);
+        final Metadata fileMetadata = mock(FileMetadata.class);
+
+        doReturn(files).when(client).files();
```

Comment:
Is there a reason to use the `doReturn().when()` instead of the recommended `when().thenReturn()` ?
We don't seem to be in a case with a void method or a spied object

Reply:
https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#doReturn(java.lang.Object)

Reply:
I guess it comes from a previous implementation, where `when(...).thenReturn(...)` was throwing an NPE. You're right that we now do not need it anymore.
Please see https://github.com/iExecBlockchainComputing/tee-worker-post-compute/pull/50/commits/5e1e7d11160814f8e64e544fe58ef3cd4690c824.

Reply:
Can't you extend it to other occurrences in the test class?

Reply:
Thanks for this update !
The other methods in the same class can be updated as well.
Exceptions can be thrown with the `when(...).thenThrow(...)` pattern

Reply:
Sure, see https://github.com/iExecBlockchainComputing/tee-worker-post-compute/pull/50/commits/82bc08bce41a8a503156605a2e5b905a32c4dfb0.